### PR TITLE
devcontainer: replace `VAULT_HOST` with `AWS_ROLE_ARN`

### PR DESCRIPTION
**Pull request by jjacobelli**
This PR is replacing the `VAULT_HOST` variable with `AWS_ROLE_ARN`. This is required to use the new token service to get AWS credentials.

### CODE REVIEW

**Review by trxcllnt — APPROVED** [review]

**Review by trxcllnt — DISMISSED** [review]
We will also need to change the base image for each devcontainer to a version that supports the new `AWS_ROLE_ARN` envvar:

```diff
-  "image": "rapidsai/devcontainers:24.04-cpp-llvm18-cuda12.0-ubuntu22.04",
+  "image": "rapidsai/devcontainers:24.12-cpp-llvm19-cuda12.0",
```

```diff
-  "image": "rapidsai/devcontainers:24.04-cpp-nvhpc24.3-ubuntu22.04",
+  "image": "rapidsai/devcontainers:24.12-cpp-nvhpc24.9",
```

File: .devcontainer/llvm18-cuda12.4/devcontainer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "image": "rapidsai/devcontainers:24.04-cpp-llvm18-cuda12.4-ubuntu22.04",
+  "image": "rapidsai/devcontainers:24.12-cpp-llvm18-cuda12.4-ubuntu22.04",
```
Comment (trxcllnt) [inline .devcontainer/llvm18-cuda12.4/devcontainer.json:2]:
```suggestion
  "image": "rapidsai/devcontainers:24.12-cpp-llvm18-cuda12.6-ubuntu22.04",
```

File: .devcontainer/nvhpc24.3/devcontainer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "image": "rapidsai/devcontainers:24.04-cpp-nvhpc24.3-ubuntu22.04",
+  "image": "rapidsai/devcontainers:24.12-cpp-nvhpc24.3-ubuntu22.04",
```
Comment (trxcllnt) [inline .devcontainer/nvhpc24.3/devcontainer.json:2]:
```suggestion
  "image": "rapidsai/devcontainers:24.12-cpp-nvhpc24.9-ubuntu22.04",
```

**Comment by trxcllnt** [comment]
Since we're updating the devcontainer files, we should also update the image tags used in CI:
* https://github.com/NVIDIA/stdexec/blob/main/.github/workflows/ci.cpu.yml#L22-L31
* https://github.com/NVIDIA/stdexec/blob/main/.github/workflows/ci.gpu.yml#L22-L30

**Review by ericniebler — APPROVED** [review]

**Comment by ericniebler** [comment]
this looks great, thanks @jjacobelli. i can't merge this while it is in the "draft" state. is this ready to go?

**Review by trxcllnt — APPROVED** [review]

**Comment by ericniebler** [comment]
thank you!

**Pull request merged**